### PR TITLE
Document new connection callback API and `useConnection` hook.

### LINKED ISSUE #471: Document new useConnection and onConnectEnd params
### Code of Conduct

- [X] I have read and agree to the xyflow contributor [Code of Conduct](https://github.com/xyflow/web/blob/main/CODE_OF_CONDUCT.md)

### Which site are you suggesting a change for?

reactflow.dev

### What content is affected by this change?

useConnection, onConnectEnd and onReconnectEnd

### What part(s) of the article would you like to see updated?

docs

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by hayleigh-dot-dev**
We tweaked the types of `useConnection`, `onConnectEnd`, and `onReconnectEnd` slightly. This PR updates the docs to reflect the new API. It also documents the `InternalNode` type as that was being referenced without being documented.

Closes #471.

### CODE REVIEW

File: sites/reactflow.dev/src/page-data/reference/ReactFlow.props.ts [hunk 1]
```diff
@@ -365,9 +365,11 @@ export const edgeEventHandlerProps: PropsTableProps = {
     },
     {
       name: 'onReconnectEnd',
-      type: '(event: React.MouseEvent, edge: Edge, handleType: "source" | "target") => void',
+      type: '(event: React.MouseEvent, edge: Edge, handleType: "source" | "target", connectionState: Omit<ConnectionState, \'inProgress\'>) => void',
```
Comment (moklick) [inline sites/reactflow.dev/src/page-data/reference/ReactFlow.props.ts:368]: We created a new type `LastConnectionType` for this. Could you use it here and for `onConnectEnd` too?

**Review by moklick — CHANGES_REQUESTED** [review]

File: sites/reactflow.dev/src/page-data/reference/ReactFlow.props.ts (hunk 1 above)
Comment (hayleigh-dot-dev) [reply sites/reactflow.dev/src/page-data/reference/ReactFlow.props.ts:368]: I can't find a reference to a `LastConnectionType` in the codebase? Both `onConnectEnd` and `onReconnectEnd` use `connectionState: FinalConnectionState` where `FinalConnectionState` is a type alias for the type documented above.

See: https://github.com/xyflow/xyflow/blob/7b8fe28b37f79dd80ed0f915b834f2f61c57b73e/packages/system/src/types/general.ts#L53 and https://github.com/xyflow/xyflow/blob/7b8fe28b37f79dd80ed0f915b834f2f61c57b73e/packages/system/src/xyhandle/types.ts#L37

---

I can make a page for `FinalConnectionState` but it felt a little superfluous to have a separate page for a type identical to `ConnectionState` save for one field. Your call.

Comment (moklick) [reply sites/reactflow.dev/src/page-data/reference/ReactFlow.props.ts:368]: yep, that make sense. If some one asks, we can still add it.

**Comment by moklick** [comment]
Good idea to document `InternalNode` as well!

**Pull request merged**